### PR TITLE
Fix architecture not supported check.

### LINKED ISSUE #7906: Architecture Arm64 not supported (code-server v3.10.2)
Bug type: General

**Description**

- VS Code Version: v1.56.1
- C/C++ Extension Version: 1.5.1
- Other extensions you installed: None

- I am running VS Code through **code-server v3.10.2** on my Android. When I installed C/C++ Extension, It gave me error that Architecture arm64 is not supported .

**Expected behavior**
Extension should work.

**Screenshots**
Screenshot_VHEditor

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Fixes https://github.com/microsoft/vscode-cpptools/issues/7906

### CODE REVIEW

File: Extension/src/main.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
 
     let errMsg: string = "";
     const arch: string = PlatformInformation.GetArchitecture();
-    if (arch !== 'x64' && (process.platform !== 'win32' || (arch !== 'x86' && arch !== 'arm64')) && (process.platform !== 'linux' || (arch !== 'x64' && arch !== 'arm' && arch !== 'arm64')) && (process.platform !== 'darwin' || arch !== 'arm64')) {
+    if (arch !== 'x64' && (process.platform !== 'win32' || (arch !== 'x86' && arch !== 'arm64')) && ((process.platform === 'win32' || process.platform === 'darwin') || (arch !== 'arm' && arch !== 'arm64')) && (process.platform !== 'darwin' || arch !== 'arm64')) {
```
Comment (bobbrow) [inline Extension/src/main.ts:41]: I find this expression too complicated to parse.  Is there a way to make it more human-readable?

**Review by Colengms — APPROVED** [review]

**Comment by Xoti-lab** [comment]
There is no ```process.platform !== 'android'``` in this?

File: Extension/src/main.ts (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/src/main.ts:41]: Yeah, but I didn't bother, i.e. pre-existing issue.

**Comment by sean-mcmanus** [comment]
> There is no `process.platform !== 'android'` in this?

It's handled by `((process.platform === 'win32' || process.platform === 'darwin') || (arch !== 'arm' && arch !== 'arm64')` -- the 1st case evaluates to false and the 2nd check of the arch applies to android or other platforms.

**Pull request merged**